lib/middleware/resource.js

```diff
@@ -4,12 +4,16 @@ function factory ({ loader }) {
   return async (req, res, next) => {
     let resources = await loader.forClassOperation(req.hydra.term)
 
+    if (resources.length > 1) {
+      return next(new Error(`no unique resource found for: <${req.hydra.term.value}>`))
+    }
```

Comment:
This check is moved up because only "multiple class resources" are an issue.

Multiple candidates for property operation can still be reconciled